pyproject.toml

```diff
@@ -1,9 +1,65 @@
-[build-system]
-requires = [
-    "setuptools>=42",
-    "wheel"
+[tool.poetry]
+name = "tidy3d-beta"
+version = "1.8.1"
+description = ""
+authors = ["Tyler Hughes <tyler@flexcompute.com>"]
+readme = "README.md"
+packages = [
+    { include = "tidy3d" },
+    { include = "tidy3d/web" },
+    { include = "tidy3d/plugins" },
 ]
-build-backend = "setuptools.build_meta"
+
+
+[tool.poetry.dependencies]
```

Comment:
I feel like we need more granular control over the dependencies like we have currently.

* `core` core packages, could be installed with `poetry install --only core`
* `web` things needed for web only
* `basic` the default install, `core` + `web` + `scipy` should be installed by default.
* `dev`should be optional, ie `poetry install --with dev`

[More info on dependency groups](https://python-poetry.org/docs/managing-dependencies/#optional-groups)
[Our current dependencies](https://github.com/flexcompute/tidy3d/tree/develop/requirements)

Reply:
It seems poetry doesn't support one group declare other groups as dependencies. We maybe thinking of an other orchestrate.